lib/graphql/schema/enum.rb

```diff
@@ -166,7 +166,11 @@ def coerce_input(value_name, ctx)
         end
 
         def inherited(child_class)
-          child_class.const_set(:UnresolvedValueError, Class.new(Schema::Enum::UnresolvedValueError))
+          if child_class.name
+            # Don't assign a custom error class to anonymous classes
+            # because they would end up with names like `#<Class0x1234>::UnresolvedValueError` which messes up bug trackers
+            child_class.const_set(:UnresolvedValueError, Class.new(Schema::Enum::UnresolvedValueError))
+          end
```

Comment:
👍 

I suspect that the same issue exists here:

https://github.com/rmosolgo/graphql-ruby/blob/v2.3.8/lib/graphql/schema/member/has_unresolved_type_error.rb#L10

Does it make sense to fix it as well?

Reply:
Ah, good call! https://github.com/rmosolgo/graphql-ruby/pull/5023